packages/core/src/wallet/metamaskBridge.ts

```diff
@@ -24,35 +28,35 @@ function isMetaMaskProvider(obj: unknown): obj is MetaMaskProvider {
 function detectMetaMaskProvider(
   windowObject: Record<string, unknown>,
   { timeout = 3000 } = {},
-) {
+): Promise<MetaMaskProvider | null> {
   let handled = false
   return new Promise<MetaMaskProvider | null>((resolve) => {
-    if (windowObject.ethereum) {
-      handleEthereum()
-    } else {
-      if (typeof windowObject.addEventListener === "function") {
-        windowObject.addEventListener("ethereum#initialized", handleEthereum, {
-          once: true,
-        })
+    const handleEIP6963Provider = (event: CustomEvent) => {
```

Comment:
thanks, 3 feedbacks:
- according to the specs, you should `Object.freeze` event objects.
- does executing `eip6963:requestProvider` inside `detectMetaMaskProvider` make sense? I'm not sure if it's the right event at the right phase.

and more generally on this listener:
why handle dapp subscription inside this bridge file instead of doing it at the main lib's entrypoint? soon it will be supported by all other providers, so it doesn't make sense to manage it here and listen on it separately per provider.

Reply:
hi @avimak 

further to above comment , i think Object.freeze is only needed from wallet side who going to announce the provider. hence in above case, we are listening an event rather than dispatch an provider announce event

And if i understand correctly, are you wanna us to make eip6963:requestProvider more generic?

as From above use case, to have eip6963: requestProvider implement in detectMetaMaskProvider is because:
- The Snap need the provider to communicate with the wallet (metamask)
- Other wallet, they inject their provider to windowObject.["starknet_xxxxxxxx"]
- The Snap is just a JS code, it is not a native wallet, it cant override anything in windowObject

Let me know if i understand the requirement correctly, we can have some counter solution to make it more simple



cc @khanti42